db/db_impl/db_impl_open.cc

```diff
@@ -1314,6 +1329,50 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
   return status;
 }
 
+void DBImpl::MoveCorruptedWalFiles(std::vector<uint64_t>& wal_numbers,
+                                   uint64_t corrupted_wal_number) {
+  // size_t corrupt_index_start = wal_numbers.size() - 1;
+  // size_t i = 0;
+  size_t num_wals = wal_numbers.size();
+
+  // Find the index of first corrupted wal.
+  size_t i = static_cast<size_t>(std::lower_bound(wal_numbers.begin(),
```

Comment:
Mixing integral vector index with iterator seems odd. Can we just use iterator throughout this function?